src/config.rs

```diff
@@ -333,6 +333,71 @@ impl Context {
     }
 }
 
+// Separate impl block for self address handling
+impl Context {
+    /// determine whether the specified addr maps to the/a self addr
+    pub(crate) async fn is_self_addr(&self, addr: &str) -> Result<bool> {
+        Ok(self
+            .get_primary_self_addr()
+            .await
+            .iter()
+            .any(|a| addr_cmp(addr, a))
```

Comment:
why an iterator here? why not just

```suggestion
            addr_cmp(addr, self.get_primary_self_addr()
```

(untested, maybe i missed sth :)

Reply:
Because we want to return `false` if no primary self addr is configured; I'll add a comment saying this. This suggestion won't complile since `get_primary_self_addr()` returns a `Result`.

What we could do is sth like:
```rust
    pub(crate) async fn is_self_addr(&self, addr: &str) -> Result<bool> {
        Ok(if let Ok(a) = self.get_primary_self_addr().await {
            addr_cmp(addr, &a)
        } else {
            false
        } || self
            .get_secondary_self_addrs()
            .await?
            .iter()
            .any(|a| addr_cmp(addr, a)))
    }
```
Do you find this easier understandable?

Reply:
you're right, your intial code is just fine. i was not really aware of the situation where no primary address is set.